lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -497,17 +497,8 @@ export class UserAgentApplication {
         acquireTokenAuthority.resolveEndpointsAsync().then(async () => {
             // On Fulfillment
             const responseType: string = isLoginCall ? ResponseTypes.id_token : this.getTokenType(account, request.scopes, false);
-            let loginStartPage: string;
 
-            if (isLoginCall) {
```

Comment:
One thing I noticed: this logic didn't run for `acquireTokenRedirect`, which doesn't seem correct. Anyone know why this was only happening for `loginRedirect`?

Reply:
This may be exactly what you mean, but `isLoginCall` is only set to true when coming in from the login APIs and not the redirect APIs, which makes sense based on the name... was maybe just being misused here

Reply:
I actually think this may be a bug, ideally we should redirect whether or not we are making a login call.

Reply:
One other thing I realized: for Angular, I don't think we would not want the redirect to occur if there is an error (so instead redirect back to the page that initiated the redirect, not the page they are trying to access). And maybe that logic should live in the Angular lib.

Reply:
Or at least the MsalGuard should be smart enough not to trigger loginRedirect if the page received an error response in the hash.

Reply:
(which would cause a login loop)

Reply:
Shouldn't that be up to the user of the lib, as it is in core? (i.e. handleResponse(), catch if error and don't call loginRedirect). The error will come back to the redirectUri same as the token response, so the logic should be the same and branch wherever the actual response handling occurs for error vs success.

Reply:
AFAIK the LOGIN_REQUEST  is cached separately <I forget the context> and since <old> angular wrapper is written a certain way, the below cache item for ANGULAR_LOGIN_REQUEST is created. It makes sense to see holistically if we need to identify login calls at all vs token acquisition calls in the cache (I remember me and Prithvi discussing this long back).

This basically narrows down to keeping `Constants.TemporaryCacheKeys.LOGIN_REQUEST` in the cache.